crates/storage/provider/src/providers/snapshot/manager.rs

```diff
@@ -452,21 +458,40 @@ impl SnapshotProvider {
         let mut provider = get_provider(range.start)?;
         let mut cursor = provider.cursor()?;
 
+        // Used as a safety check to make sure we don't get stuck on a loop on an unexpected issue
+        // or corrupted file.
+        //
+        // Since we're looping through a range, we should retry at most once for a specific
+        // **number: if we're changing files (get_provider) aka checking the lower block range file.
+        let mut retrying = false;
+
         // advances number in range
         'outer: for number in range {
             // advances snapshot files if `get_fn` returns None
             'inner: loop {
                 match get_fn(&mut cursor, number)? {
                     Some(res) => {
+                        retrying = false;
+
                         if !predicate(&res) {
                             break 'outer
                         }
+
                         result.push(res);
                         break 'inner
                     }
                     None => {
+                        if retrying {
+                            let err = if segment.is_headers() {
```

Comment:
can we add a warn! here?

Reply:
we can, but this is not an actual malfunction, which I kinda associate with a warning.

if the range spans across multiple files, there will be a time when the `number` no longer belongs to the current file (and thus returning `None` on the `get_fn`), and we need to find the associated file provider (`get_provider`).

Reply:
oh wait you mean inside the `if retrying`, yeah okay adding